cloud/wfm/src/executor/ExecutorImpl.ts

```diff
@@ -0,0 +1,37 @@
+import * as Promise from 'bluebird';
+import {Process} from '../process';
+import {ProcessInstance} from '../process-instance';
+import {Task, TaskEventData, TaskStatus} from '../task';
+import {Executor} from './index';
+import {InstanceRepository} from './InstanceRepository';
+
+export {InstanceRepository};
+export default class ExecutorImpl implements Executor {
+  public instance: ProcessInstance;
+  constructor(public process: Process, public instanceRepository: InstanceRepository) {
+    this.instance = this.process.createInstance();
+  }
+  public start() {
+    this.runCurrentTask();
+  }
+  protected onTaskDone(e: TaskEventData<Task>) {
```

Comment:
`onTaskDone` here is just an implementation detail for the executor for it to track the state of the instance's tasks, hence `protected`. Maybe we can set it up to forward the events, but separately.

I used 'done' because that's the name of the state, do you think we should change it?

Going to try out the `TaskEventListener` idea :+1: .